### PR TITLE
Add keymap

### PR DESCRIPTION
**Pull request by NejcZdovc**
Implementing key map add screen. More info here #78.

### CODE REVIEW

**Comment by fjozsef** [comment]
What is the progress of this? Can I review?

**Comment by NejcZdovc** [comment]
No, we need to talk about this, because we need animation on key maps here as well. We would need to refactor KeymapComponent so that will be a separate component that will already include animations and layers at top. What was done so far is basically everything except animations and current list length. Will ping you tomorrow on chat to discus it further.

**Comment by NejcZdovc** [comment]
Working on #82 to separate components that can be then used here.

**Comment by NejcZdovc** [comment]
Everything is now done except text in the tool tip. I was checking out how it's done in popover and I think that we need helper (maybe I just didn't find it) that will took keyAction in and return all properties (key action, scan code, long press, modifiers). @fjozsef Do we have something like this?

File: src/components/keymap/add/keymap-add.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { Keymap } from '../../../config-serializer/config-items/Keymap';
+import { Keymaps } from '../../../config-serializer/config-items/Keymaps';
+import { SvgKeyboardComponent } from '../../svg/keyboard/svg-keyboard.component';
+import { UhkConfigurationService } from '../../../services/uhk-configuration.service';
+
+@Component({
+    directives: [SvgKeyboardComponent],
```
Comment (fjozsef) [inline src/components/keymap/add/keymap-add.component.ts]: Move to app.module

File: src/components/keymap/add/keymap-add.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    providers: [UhkConfigurationService],
```
Comment (fjozsef) [inline src/components/keymap/add/keymap-add.component.ts]: Order: selector, template, styles, providers

File: src/components/keymap/add/keymap-add.component.ts (hunk 1 above)
Comment (NejcZdovc) [inline src/components/keymap/add/keymap-add.component.ts]: Done

File: src/components/keymap/add/keymap-add.component.ts (hunk 2 above)
Comment (NejcZdovc) [inline src/components/keymap/add/keymap-add.component.ts]: Done

File: src/components/svg/wrap/svg-keyboard-wrap.component.scss [hunk 3]
```diff
@@ -10,12 +10,34 @@ svg-keyboard {
     max-width: 1400px;
     position: absolute;
     left: 0;
-    transform: translateX(-100%);
+    transform: translateX(-101%);
```
Comment (fjozsef) [inline src/components/svg/wrap/svg-keyboard-wrap.component.scss]: Why did you changed it?

Comment (NejcZdovc) [inline src/components/svg/wrap/svg-keyboard-wrap.component.scss]: It was changed, because there where some traces of keyboard that supposed to be hidden.

**Comment by mondalaci** [comment]
Am I right that `all-keymaps.json` contains the keymaps shown on the add keymaps page? If so, I think `preset-keymaps.json` would be a better name. Even better, how about creating a `preset-keymaps` directory and putting files per keymaps into it like `vim.json`? Then the build process would merge these files to a single file. It'd result in much better maintainability over the long term.

**Comment by NejcZdovc** [comment]
@fjozsef wanted to rename all to default, so it was just renamed :smile:

**Pull request merged**